src/scratchOrgInfoApi.ts

```diff
@@ -0,0 +1,528 @@
+  timeout?: number;
+  delay?: number;
+}): Promise<AuthInfo> => {
+  const logger = await Logger.child('getAuthInfo');
+
+  const retryAuthorize = retryDecorator(async (opts: AuthInfo.Options): Promise<AuthInfo> => AuthInfo.create(opts), {
+    timeout: options.timeout,
+    delay: options.delay,
+    retries: options.retries,
+  });
+
+  if (options.retries) {
+    try {
+      return await retryAuthorize({
+        username: options.username,
+        parentUsername: options.hubOrg.getUsername(),
+        oauth2Options: options.oauth2Options,
+      });
+    } catch (err) {
+      const error = err as RetryError;
+      logger.error(error);
+      if (error.message.startsWith('Timeout after')) {
+        throw SfdxError.create('salesforce-alm', 'org_create', 'jwtAuthRetryTimedOut', [
+          options.username,
+          options.timeout,
+          options.retries,
+        ]);
+      }
+      throw error.lastError || error;
+    }
+  } else {
+    try {
+      return await AuthInfo.create({
+        username: options.username,
+        parentUsername: options.hubOrg.getUsername(),
+        oauth2Options: options.oauth2Options,
+      });
+    } catch (error) {
+      throw SfdxError.wrap(error as Error);
+    }
+  }
+};
+
+/**
+ * after we successfully signup an org we need to trade the auth token for access and refresh token.
+ *
+ * @param scratchOrgInfoComplete - The completed ScratchOrgInfo which should contain an access token.
+ * @param hubOrg - the environment hub org
+ * @param authInfo - The AuthInfo object
+ * @param setAsDefault - {boolean} - whether to save this org as the default for this workspace.
+ * @param alias - scratch org alias
+ * @returns {Promise<void>}
+ */
+const saveAuthInfo = async (options: {
+  scratchOrgInfoComplete: ScratchOrgInfo;
+  hubOrg: Org;
+  authInfo: AuthInfo;
+  setAsDefault: boolean;
+  alias?: string;
+}): Promise<void> => {
+  const logger = await Logger.child('saveAuthInfo');
+
+  await options.authInfo.save({
+    devHubUsername: options.hubOrg.getUsername(),
+    created: new Date(options.scratchOrgInfoComplete.CreatedDate ?? new Date()).valueOf().toString(),
+    expirationDate: options.scratchOrgInfoComplete.ExpirationDate,
+    clientId: options.scratchOrgInfoComplete.ConnectedAppConsumerKey,
+    createdOrgInstance: options.scratchOrgInfoComplete.SignupInstance,
+    isDevHub: false,
+    snapshot: options.scratchOrgInfoComplete.Snapshot,
+  });
+
+  if (options.alias) {
+    logger.debug(`_authorize - setting alias to ${options.alias}`);
+    await options.authInfo.setAlias(options.alias);
+    logger.debug(`_authorize - AuthInfo has alias to ${options.authInfo.getFields().alias}`);
+  }
+  if (options.setAsDefault) {
+    await options.authInfo.setAsDefault({ defaultUsername: true });
+  }
+
+  logger.debug(`_authorize - orgConfig.loginUrl: ${options.authInfo.getFields().loginUrl}`);
+  logger.debug(`_authorize - orgConfig.instanceUrl: ${options.authInfo.getFields().instanceUrl}`);
```

Comment:
All of these debugs should not have "_authorize - " in them.  That is left over from toolbelt.

You probably only need 1 debug of the alias (line 273) as well.